File: HelloShoesBE/src/main/java/lk/ijse/HelloShoesBE/entity/Inventory.java
package lk.ijse.HelloShoesBE.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.HashSet;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Table(name = "inventory")
@Entity
public class Inventory implements SuperEntity{
    @Id
    private String itemCode;
    private String itemDesc;
    @Column(columnDefinition = "LONGTEXT")
    private String itemPic;
    private String category;
    private String supplierCode;
    private String supplierName;
    private double buyPrice;
    private double sellPrice;
    private double profit;
    private double profitMargin;

    @OneToMany(mappedBy = "inventory")
    private Set<SaleInventories> saleInventories = new HashSet<>();

    @OneToMany(mappedBy = "inventory")
    private Set<SupplierInventories> supplierInventories = new HashSet<>();
}

File: HelloShoesFE/model/Inventory.js
export class Inventory{
    constructor(itemCode, itemDesc, itemPic, category, supplierCode, supplierName, buyPrice, sellPrice, profit, profitMargin) {
        this.itemCode = itemCode;
        this.itemDesc = itemDesc;
        this.itemPic = itemPic;
        this.category = category;
        this.supplierCode = supplierCode;
        this.supplierName = supplierName;
        this.buyPrice = buyPrice;
        this.sellPrice = sellPrice;
        this.profit = profit;
        this.profitMargin = profitMargin;
    }
}